### PR TITLE
Allows turrets to shoot cyborgs.

### PR DESCRIPTION
**Pull request by Y0SH1M4S73R**
:cl: Y0SH1 M4S73R (with credit to Anonmare)
add: Turrets can target silicons. Whether or not they will can be configured in the turret's interface.
fix: Syndicate turrets will shoot at non-Syndicate silicons by default.
/:cl:

Basically #32938 but with working code.

### CODE REVIEW

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -62,6 +62,7 @@
 	var/auth_weapons = 0	//checks if it can shoot people that have a weapon they aren't authorized to have
 	var/stun_all = 0		//if this is active, the turret shoots everything that isn't security or head of staff
 	var/check_anomalies = 1	//checks if it can shoot at unidentified lifeforms (ie xenos)
+	var/target_silicons = 0 //checks if it can shoot silicons
```
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: FALSE

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 2]
```diff
@@ -550,6 +559,7 @@
 	faction = "syndicate"
 	emp_vunerable = 0
 	desc = "A ballistic machine gun auto-turret."
+	target_silicons = 1
```
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: TRUE

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 3]
```diff
@@ -396,6 +400,11 @@
 			if(M.occupant && !in_faction(M.occupant))
 				if(assess_perp(M.occupant) >= 4)
 					targets += M
+					
+		if(issilicon(A))
+			var/mob/living/silicon/S = A
+			if(!in_faction(C) && target_silicons) //if the target is a silicon and not in our faction, target it if we are allowed to target silicons
```
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: check target_silicons first

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by vuonojenmustaturska** [comment]
While you're rooting around this bit of code, please change the 0/1s to TRUE/FALSE, and refactor the check to be `if (target_silicons && issilicon(A))`, that way the branch is skipped early when the setting is false, which seems to be the default.

**Comment by WJohn** [comment]
These syndie turrets don't shoot at syndicate borgs, right? Faction overrules that, yes?

**Comment by Anonmare** [comment]
@WJohn I think it might, non-AI turrets actually consider people wearing Agent IDs a non-threat even if they normally would be. If not, it shouldn't be too difficult to add a check that looks for an Agent ID

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 4]
```diff
@@ -396,6 +400,11 @@
 			if(M.occupant && !in_faction(M.occupant))
 				if(assess_perp(M.occupant) >= 4)
 					targets += M
+					
+		if(issilicon(A))
+			var/mob/living/silicon/S = A
+			if(target_silicons && !in_faction(S)) //if the target is a silicon and not in our faction, target it if we are allowed to target silicons
```
Comment (AnturK) [inline code/game/machinery/porta_turret/portable_turret.dm]: Still can be moved before the first type check.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Pull request closed without merging**